src/components/AddComment.jsx

```diff
@@ -0,0 +1,78 @@
+import { useState } from "react";
+import { postComment } from "../api";
+
+const AddComment = ({ article_id, setArticleComments }) => {
+  const [newComment, setNewComment] = useState("");
+  const [username, setUsername] = useState("");
+  const [addButton, setAddButton] = useState(false);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    postComment(newComment, article_id, username)
+      .then((newComment) => {
+        setNewComment("");
+        setUsername("");
+        setArticleComments((currComments) => {
+          const newCommentList = [...currComments];
+          newCommentList.unshift(newComment);
+          return newCommentList;
+        });
+        alert("Comment added!");
```

Comment:
Stay away from alerts, as you wouldn't really expect these on a real site. Could you turn this into a text display under the comment box that displays the validation or error message depending on whether the request was successful?